alexandria/core/management/commands/fix_copied_file_references.py

```diff
@@ -0,0 +1,90 @@
+import botocore
+from django.core.management.base import BaseCommand
+from tqdm import tqdm
+
+from alexandria.core.models import File
+
+
+class Command(BaseCommand):
+    help = "Fixes copied file references by creating a new copy."
+
+    def add_arguments(self, parser):
+        parser.add_argument("--dry", dest="dry", action="store_true", default=False)
+
+    def handle(self, *args, **options):
+        filename_counts = {}
+        for tmp_file in tqdm(
+            File.objects.filter(variant="original").order_by("created_at")
+        ):
+            filename = tmp_file.content.name
+            if filename in filename_counts:
+                filename_counts[filename].append(tmp_file.pk)
+            else:
+                filename_counts[filename] = [tmp_file.pk]
+
+        fix_count = 0
+        for filename, pks in filename_counts.items():
+            # Skip files that only have one reference.
+            if len(pks) <= 1:
+                continue
+
+            # Find duplicates, which are files with the same filename and no checksum.
+            duplicates = File.objects.filter(pk__in=pks).order_by("created_at")
+            checksum = duplicates.first().checksum
+
+            for duplicate in duplicates:
+                if duplicate.checksum != checksum:
+                    self.stderr.write(
+                        self.style.ERROR(
+                            f"Failed: File {duplicate.pk} with filename {filename} has a different checksum than the original."
+                        )
+                    )
+                    continue
+
+                current_filename = duplicate.content.name
+                new_filename = f"{duplicate.pk}_{duplicate.name}"
+
+                # Only fix copied files, skip the original
+                if current_filename == new_filename:
+                    continue
```

Comment:
This seems a bit fragile, no? You already know that the original is the first one in the duplicates QS, so why not make use of that fact instead?

Maybe something along the lines of `original, *duplicates = File.objects.filter(...)`

Reply:
Changed to exclude the original from the collection in the for loop